src/mixins.coffee

```diff
@@ -64,3 +77,95 @@ Ember.Widgets.BodyEventListener = Ember.Mixin.create
   _removeDocumentHandlers: ->
     $(@get('bodyElementSelector')).off "click", @_clickHandler
     @_clickHandler = null
+
+Ember.Widgets.TabbableModal = Ember.Mixin.create Ember.Widgets.KeyboardHelper,
+  # these are variables from the modal component (modal.coffee) to enforce
+  # the modality when clicking outside of the modal and to allow pressing ESC
+  # to cancel the modal. Please check if this mixin is going to be used
+  # outside of modal.coffee
+  enforceModality:  no
+  escToCancel:      yes
+
+  currentFocus: null
+
+  _focusTabbable: ->
+     # Set focus to the first match:
+     # 1. The saved focused element
+     # 2. First element inside the dialog matching [autofocus]
+     # 3. Tabbable element inside the content element
+     # 4. The close button (has class "close")
+    hasFocus = [@get 'currentFocus']
+    unless @$().has(hasFocus[0]).length
+      hasFocus = @$('[autofocus]')
+    if hasFocus.length == 0
+      hasFocus = @$(':tabbable')
+    if focusElement = hasFocus[0]
+      if focusElement.className.indexOf('close') > -1
+        # if we have more than two tabbable objects, we do not want to tab
+        # to close button
+        # while if we do not have any choice, the close button is chosen
+        if hasFocus.length > 1
+          focusElement = hasFocus[1]
+      focusElement.focus()
+      @set 'currentFocus', focusElement
+
+  _checkContainingElement: (containers, element) ->
+    for containerItem in containers
+      if containerItem is element or $.contains(containerItem, element)
+        return yes
+    return no
```

Comment:
@igillis did you mean the default returned value will be `no`? Should we make it more explicit for easier reading?